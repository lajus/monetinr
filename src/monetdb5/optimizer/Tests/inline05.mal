# remapping inline functions
# the crucial part is to recognize valid inlines
function foo{inline}(a:int,b:int):int;
	c:= a+b;
	io.print(c); # this has a memory-less side effect
	return foo:=1;
end foo;

function qry();
	b:= bat.new(:oid,:int);
	bat.insert(b,0@0,1);
	z:bat[:oid,:int]:= mal.multiplex("user","foo",b,b);
	z:bat[:oid,:dbl]:= mal.multiplex("user","foo",b,2);
end qry;

	optimizer.inline("user","qry");
	optimizer.remap("user","qry");
	mdb.List("user","qry");
