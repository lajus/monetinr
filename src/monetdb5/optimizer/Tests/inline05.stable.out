stdout of test 'inline05` in directory 'monetdb5/optimizer` itself:


# 18:00:44 >  
# 18:00:44 >   mserver5 "--config=/ufs/mk/monet5/Linux/etc/monetdb5.conf" --debug=10 --set "monet_mod_path=/ufs/mk/monet5//Linux/lib/MonetDB5:/ufs/mk/monet5//Linux/lib/MonetDB5/lib:/ufs/mk/monet5//Linux/lib/MonetDB5/bin" --set "gdk_dbfarm=/ufs/mk/monet5//Linux/var/MonetDB5/dbfarm" --set "sql_logdir=/ufs/mk/monet5//Linux/var/MonetDB5/sql_logs"  --set mapi_open=true --set xrpc_open=true --set mapi_port=34223 --set xrpc_port=43286 --set monet_prompt= --trace --dbname=mTests_src_optimizer  inline05.mal
# 18:00:44 >  

# MonetDB server v5.3.0, based on kernel v1.21.0
# Serving database 'mTests_src_optimizer'
# Compiled for x86_64-redhat-linux-gnu/64bit with 64bit OIDs dynamically linked
# Copyright (c) 1993-2007 CWI, all rights reserved
# Visit http://monetdb.cwi.nl/ for further information
# Listening for connection requests on mapi:monetdb://eir.ins.cwi.nl:32348/
function user.foo{inline}(a:int,b:int):int;
    c := calc.+(a,b);
    io.print(c);
# this has a memory-less side effect 
    return foo := 1;
end foo;
function user.qry():void;
    b := bat.new(:oid,:int);
    bat.insert(b,0@0,1);
    z:bat[:oid,:dbl]  := mal.multiplex("user","foo",b,b);
    z:bat[:oid,:dbl]  := mal.multiplex("user","foo",b,2);
end qry;
function user.main():void;
# remapping inline functions 
# the crucial part is to recognize valid inlines 
    mdb.List("user","qry");
end main;
function user.qry():void;               #  0 qry:void := user.qry() {G}
    b := bat.new(:oid,:int);            #  1 b:bat[:oid,:int] := CMDBATnew(_2:oid, _3:int) {G}
    bat.insert(b,0@0,1);                #  2 _4:bat[:oid,:int] := BKCinsert_bun(b:bat[:oid,:int], _5:oid, _6:int) {G}
    z:bat[:oid,:dbl] {inline} := mal.multiplex("user","foo",b,b);#  3 z:bat[:oid,:dbl] := OPTremapMultiplex(_8:str, _9:str, b:bat[:oid,:int], b:bat[:oid,:int]) {G}
    z:bat[:oid,:dbl] {inline} := mal.multiplex("user","foo",b,2);#  4 z:bat[:oid,:dbl] := OPTremapMultiplex(_8:str, _9:str, b:bat[:oid,:int], _10:int) {G}
end qry;                                #  5  

# 18:00:44 >  
# 18:00:44 >  Done.
# 18:00:44 >  

